### DISCUSSION TITLE
How to set cosine lr scheduler?

### DISCUSSION THREAD

**Discussion by lucasjinreal** [facebookresearch/detectron2#3296] [Q&A]
I try train YOLOv3 single stage model, the lr hardly converge on coco. I am also adopt all YOLOX model define and hard to make it converge on coco.

But lr can work on custom dataset (50k images with 4 classes).

SO i am think of it's because of cosine lr sheduler problem or not, is there a standered way to open cosine lr scheduler? 

I know there is a `WarmupCosineLR` but as detectron2 official says it was deprecated. What's the way to specific it now?

**Comment by ppwwyyxx** [comment]
If you're using yacs/yaml config, it can be set by SOLVER.LR_SCHEDULER_NAME="WarmupCosineLR"

**Reply by lucasjinreal** [reply]
thanks, solved this.
